### PR TITLE
Fix newly enforced package:pedantic lints

### PR DESCRIPTION
**Pull request by natebosch**
- always_declare_return_types
- annotate_overrides
- omit_local_variable_types
- prefer_single_quotes
- unnecessary_this

### CODE REVIEW

File: lib/src/int32.dart [hunk 1]
```diff
@@ -54,12 +54,12 @@ class Int32 implements IntX {
...
   static Int32 parseRadix(String s, int radix) {
     _validateRadix(radix);
-    Int32 x = ZERO;
-    for (int i = 0; i < s.length; i++) {
-      int c = s.codeUnitAt(i);
-      int digit = _decodeDigit(c);
+    var x = ZERO;
+    for (var i = 0; i < s.length; i++) {
+      var c = s.codeUnitAt(i);
+      var digit = _decodeDigit(c);
```
Comment (cbracken) [inline lib/src/int32.dart]: The elimination of the types here make the code less readable.

File: lib/src/int32.dart [hunk 2]
```diff
@@ -259,64 +273,80 @@ class Int32 implements IntX {
...
     if (other is Int64) {
-      return this.toInt64() >= other;
+      return toInt64() >= other;
     }
     return _i >= _toInt(other);
   }
 
+  @override
   bool get isEven => (_i & 0x1) == 0;
+  @override
```
Comment (cbracken) [inline lib/src/int32.dart]: nit: Would you mind adding a blank line between these getters?

**Comment by natebosch** [comment]
> Consider adding an analysis_options.yaml file and an analyzer test to enforce these.

They should already be enforced.

https://github.com/dart-lang/fixnum/blob/8c6f0df52a6dc92293441ba1545b5b669543a08b/analysis_options.yaml#L1

> Yes to the single quotes and `@override` annotations. No to `var`.

This was a choice made globally internally. Technically we _can_ decide to ignore specifically `omit_local_variable_types` for this package, but it's not a precedent that I'm happy to set...

**Comment by cbracken** [comment]
> They should already be enforced.

Ah awesome.

>> Yes to the single quotes and @override annotations. No to var.

> This was a choice made globally internally. Technically we can decide to ignore specifically omit_local_variable_types for this package, but it's not a precedent that I'm happy to set...

Most code will be read far more than it's ever written. The lint in question has no effect on correctness, it's entire purpose is readability-oriented. In this case, the resulting code is less readable. I don't think there's any point in enforcing lints blindly -- they should be applied where they result in an improvement, or at least where they don't cause the opposite of their intended effect.

That said -- for the most part, I no longer really work on this code, so I leave it up to the current maintainers judgement.

File: lib/src/int32.dart (hunk 2 above)
Comment (rakudrama) [reply lib/src/int32.dart]: +1. Once the methods become multi-line due to the annotation they need something to separate them.

File: lib/src/int64.dart [hunk 3]
```diff
@@ -70,18 +70,18 @@ class Int64 implements IntX {
       throw FormatException("No digits in '$s'");
     }
 
-    int d0 = 0, d1 = 0, d2 = 0; //  low, middle, high components.
+    var d0 = 0, d1 = 0, d2 = 0; //  low, middle, high components.
```
Comment (rakudrama) [inline lib/src/int64.dart]: Please leave `int` everywhere in this file. This code is performance critical and has been broken before by well-meaning changes that accidentally generalize the types, resulting in large regressions of protobuf parsing.

File: lib/src/int64.dart [hunk 4]
```diff
@@ -915,20 +957,20 @@ class Int64 implements IntX {
...
 
-      const double K2 = 17592186044416.0; // 2^44
-      const double K1 = 4194304.0; // 2^22
+      const K2 = 17592186044416.0; // 2^44
+      const K1 = 4194304.0; // 2^22
 
       // Approximate double values for [a] and [b].
-      double ad = a0 + K1 * a1 + K2 * a2;
-      double bd = b0 + K1 * b1 + K2 * b2;
+      var ad = a0 + K1 * a1 + K2 * a2;
```
Comment (rakudrama) [inline lib/src/int64.dart]: I believe the inferred type in most of these is `num`, which is polymorphic in the operations, so not the same as `double`. We might have to go in later with nnbd and change this to have `as double`.

Comment (natebosch) [reply lib/src/int64.dart]: I don't believe the lint is intended to fire if the type annotation doesn't exactly match the inferred type.

Both of these are inferred as `double` today.

**Comment by natebosch** [comment]
> Please do not replace `int` and `double` with `var` in `int32.dart` and `int64.dart`.

I won't be pushing on any PRs that pick and choose which `package:pedantic` lints to honor. If you'd like to unblock travis without following these lints there are a couple paths forward:

- Best option: fix the other newly enforced lints and exclude `omit_local_variable_types`:
  ```yaml
  analyzer:
    errors:
      omit_local_variable_types: ignore
  ```
- Pin to an older version of `package:pedantic` and don't enforce any new lints:
  ```yaml
  include: package:pedantic/analysis_options.1.8.0.yaml
  ```

> Is it possible to, for these files, to enforce the opposite, that no int or double local is var?

That doesn't match any lint I'm aware of. If you want to push on it you could ask the analyzer team for a new lint, or perhaps that `int` and `double` are hardcoded exclusions to `omit_local_variable_types`.

**Pull request closed without merging**